### PR TITLE
DRUP-640: #ready rework of blog migration Squashed commit of the following:

### PR DESCRIPTION
**Pull request by z3cka**
commit 88d80f6b0663f353bc507287a013a3136b95e5b6
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Tue Jun 7 16:13:39 2016 -0700

    upping to 3.0-beta1 for deployment to www-test

commit 26a7a2617fa09339b165ddfd8015c6dbc7a5b3de
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Tue Jun 7 16:08:24 2016 -0700

    updating to current version of bookplates in upstream master

commit facc9f363327ca75eb66bc223004cdcfa229e709
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Tue Jun 7 21:54:40 2016 +0000

    adds ligic for blog archives path and create alias

commit a87da169c25a834735c4c48b75e6181250d0dad5
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Mon Jun 6 23:29:28 2016 +0000

    renders the link in /taxonomy/term/%blog_name/archives/%year/%month (YYYY/MM) format

commit e5567e4405330bcb567a60a88225972ec63049f2
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Mon Jun 6 23:10:54 2016 +0000

    adds blog views argument handling for archives and missing blog_cattag page manager panel

commit 7bd31e3f0b42b012ef29c4dad62a04ef23f6806e
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Wed Jun 1 22:52:27 2016 +0000

    adds some blog_content_pane[s] and logic for generating cattag aliases on demand

commit b5b19c2e7c44178e1e23be1cac47e266fea5607a
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Wed May 25 23:36:12 2016 +0000

    removes unused hook_menu() and preprocesses cats and tags for filter linking from sidebar

commit 53015164c5e3545ae93bee25104ed264f0e6e1b7
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Mon May 23 20:12:19 2016 +0000

    fixes blog tag query and adds related panelizer configs

commit e1de48e74a334fd76b8cb2f8eb22c9c0ec822827
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Mon May 23 17:06:32 2016 +0000

    fixes blog panel passing proper args to view pane for displaying blog (news) categories

commit 86fdccddbd0113933033b8083e597d0393ceb61c
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu May 19 23:28:42 2016 +0000

    adds logic to create path alias for blog nodes visited if it does not exist

commit bd3114463d3053a8a377af2f48cd1edc391f45cf
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Fri May 6 22:33:50 2016 +0000

    WIP path stuff for blogs

commit 2bcc09728ea540b4d6777e6b717647ccd0424a0b
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Fri May 6 20:27:46 2016 +0000

    adds redirect blog nodes to new internal path structure: /taxonomy/term/%/node/%

commit 3ae54f332c247a8ad3820f355092667c4687cd2e
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Fri May 6 19:05:57 2016 +0000

    removes blog post node view variant

commit 820775e0cdbc96ebf451804b6b20ee13c8c698fd
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu May 5 23:26:20 2016 +0000

    WIP on blog node redirect and alias workflow

commit ce9bda4bd3373421692c7f70e683bc3a28fd842f
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu May 5 16:29:28 2016 +0000

    adds default panelizer setting to Blog Name taxonomy

commit 4ac07aae9e72f4ff4bbe68374e561f2865b80557
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu Apr 28 23:20:50 2016 +0000

    adds blog and post handlers via panelizer, etc

commit 11663360e3cae12377f2dbdec7f16b8a72c9512b
Merge: d2878f9 f39dd34
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Wed Apr 27 00:25:12 2016 +0000

    Merge remote-tracking branch 'remotes/origin/wpm' into wpm_rb160422

    Conflicts:
    	www/sites/all/modules/features/uclalib_news/uclalib_news.info
    	www/sites/all/modules/features/uclalib_news/uclalib_news.panelizer.inc

commit d2878f9a99e2cfe0ce470a48b5e0cf1e2dcc993a
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu Apr 14 18:01:50 2016 +0000

    replaces blog panel content with pane versions of views displays

commit a90b172b85ac23aa8d5724199723d4e91299b4f3
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu Apr 14 17:01:40 2016 +0000

    adds layout and content configs for panelized blog_name terms

commit 0e2eba58a6c9990059100bc23cff09e67bdad917
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Tue Apr 12 17:01:04 2016 +0000

    removes conflicting 'term exists' panel variant; was blocking blog related selection rules

commit acebd218e3d4911596a4c7f55a96c56b694397ed
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Tue Apr 12 16:39:14 2016 +0000

    adds taxonomy_term:blog_name panelizer settings

commit f39dd340616d6e6613d0bd070b0b5ce75526b5eb
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Sat Apr 16 06:13:02 2016 +0000

    adds Blog Content - Single views display to show single post limited by selection rules

commit ce0d14a758f90d34c3be089071c8208cb64fe2e0
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu Apr 14 18:01:50 2016 +0000

    replaces blog panel content with pane versions of views displays

commit 838d01f7a7b32d105fe99dff419e38ee9178a2b5
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Thu Apr 14 17:01:40 2016 +0000

    adds layout and content configs for panelized blog_name terms

commit 96a96c0878b5137f161bd256db8c61ff6d4443c1
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Tue Apr 12 17:01:04 2016 +0000

    removes conflicting 'term exists' panel variant; was blocking blog related selection rules

commit 664f2ae97a29b9e6e4c79183ce1e606a82291e13
Author: Casey Grzecka <cgrzecka@library.ucla.edu>
Date:   Tue Apr 12 16:39:14 2016 +0000

    adds taxonomy_term:blog_name panelizer settings

### CODE REVIEW

File: www/sites/all/modules/features/uclalib_news/uclalib_news.module [hunk 1]
```diff
@@ -19,6 +18,94 @@ function uclalib_news_form_views_exposed_form_alter(&$form, &$form_state) {
...
+        // look for desired path
+        $path_alias_search = drupal_get_path_alias('taxonomy/term/'. $node->field_news_blog_name[LANGUAGE_NONE][0]['tid'] .'/node/'. $node->nid);
+        // create path alias if it does not exist
+        if (current_path() == request_path() && $path_alias_search == 'taxonomy/term/'. $node->field_news_blog_name[LANGUAGE_NONE][0]['tid'] .'/node/'. $node->nid) {
+          // check if there is a blog base path
+          $blog_root_path = drupal_get_path_alias('taxonomy/term/'. $node->field_news_blog_name[LANGUAGE_NONE][0]['tid']);
+          // format date for use in URL
+          $post_date = format_date($node->created, 'custom', 'Y/m/d');
+          // manipulate title string to be url friendly
+          $post_title_path = strtolower($node->title);
```
Comment (akohler) [inline www/sites/all/modules/features/uclalib_news/uclalib_news.module]: @z3cka You use `$node->field_news_blog_name[LANGUAGE_NONE][0]['tid']` 3 times in this function - assign it to a local variable.

File: www/sites/all/modules/features/uclalib_news/uclalib_news.module [hunk 2]
```diff
@@ -19,6 +18,94 @@ function uclalib_news_form_views_exposed_form_alter(&$form, &$form_state) {
...
+}
+
+/**
+ * Implements hook_taxonomy_term_view().
+ */
+function uclalib_news_taxonomy_term_view($term, $view_mode, $langcode) {
+  // check if it's a blog
+  if ($term->vocabulary_machine_name == 'blog_name') {
+    // check if it is a category or tag in requested blog
+    if (arg(3) == 'cattag') {
```
Comment (akohler) [inline www/sites/all/modules/features/uclalib_news/uclalib_news.module]: I don't understand use of `arg(3)` here.  I see 3 arguments passed to `uclalib_news_taxonomy_term_view` - is it one of these, or if not, where does `arg(3)` come from?

File: www/sites/all/modules/features/uclalib_news/uclalib_news.module [hunk 3]
```diff
@@ -19,6 +18,94 @@ function uclalib_news_form_views_exposed_form_alter(&$form, &$form_state) {
...
+      // check if the current path needs and alias
+      if (current_path() == request_path()) {
+        // get blog's root path
+        $blog_root_path = drupal_get_path_alias('taxonomy/term/'. $term->tid);
+        // get cattag path
+        $cattag_path = drupal_get_path_alias('taxonomy/term/'. arg(4));
```
Comment (akohler) [inline www/sites/all/modules/features/uclalib_news/uclalib_news.module]: `arg(4)` is another undocumented(?) value.

File: www/sites/all/modules/features/uclalib_news/uclalib_news.module [hunk 4]
```diff
@@ -19,6 +18,94 @@ function uclalib_news_form_views_exposed_form_alter(&$form, &$form_state) {
...
+        // prepare the new alias
+        $path = array(
+          'source' => drupal_get_path_alias(),
+          'alias' => $blog_root_path .'/'. $cattag_path,
+        );
+        dpm($path, "\$path");
```
Comment (akohler) [inline www/sites/all/modules/features/uclalib_news/uclalib_news.module]: Remove `dpm` before deployments.

File: www/sites/all/modules/features/uclalib_news/uclalib_news.module [hunk 5]
```diff
@@ -19,6 +18,94 @@ function uclalib_news_form_views_exposed_form_alter(&$form, &$form_state) {
...
+        // save and go to the new alias
+        path_save($path);
+        drupal_goto(drupal_get_path_alias());
+      }
+    }
+    // check if it is an archives path and create alias
+    if (arg(3) == 'archives') {
```
Comment (akohler) [inline www/sites/all/modules/features/uclalib_news/uclalib_news.module]: Another use of `arg(3)` - define a local variable?

File: www/sites/all/modules/features/uclalib_news/uclalib_news.module [hunk 6]
```diff
@@ -19,6 +18,94 @@ function uclalib_news_form_views_exposed_form_alter(&$form, &$form_state) {
...
+      // check if the current path needs an alias
+      if (current_path() == request_path()) {
+        // get blog's root path
+        $blog_root_path = drupal_get_path_alias('taxonomy/term/'. $term->tid);
+        // get year and month for building path
+        $year_month = array_slice(arg(), 4);
```
Comment (akohler) [inline www/sites/all/modules/features/uclalib_news/uclalib_news.module]: See earlier comments about use of `arg()`.

**Comment by z3cka** [comment]
awesome; github shows @akohler's comments as outdated! here's the new changes: c0e3808 **will squash these and prep new PR** _if these changes look good_.

**Comment by akohler** [comment]
@z3cka This looks great to me, easier to understand the code's purpose with these improvements.  Re-squash and deploy away!

**Comment by z3cka** [comment]
excellent. time to squash and add a tweak to the byline for blog posts _(forgot about that small change)_.

**Pull request closed without merging**